jfr-mappers/src/main/java/com/newrelic/jfr/tosummary/AbstractThreadDispatchingSummarizer.java

```diff
@@ -17,8 +17,14 @@
   protected final Map<String, EventToSummary> perThread = new HashMap<>();
 
   @Override
-  public Stream<Summary> summarizeAndReset() {
-    return perThread.values().stream().flatMap(EventToSummary::summarizeAndReset);
+  public Stream<Summary> summarize() {
+    Stream<Summary> summaryStream = perThread.values().stream().flatMap(EventToSummary::summarize);
+    return summaryStream;
```

Comment:
you may be wondering why we can't just clear the map here after we return the stream of summaries. That would be ideal because then we wouldn't have to change anything else. The reason why we can't is because the stream operates on the map, so if we clear the map before the stream is used, the stream will return no summaries because they've all been cleared from the map. So the work around is to create a separate method to clear the map after the stream is read.

Reply:
Inline the variable to make this a one-liner.